bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/bridge/slip/FunctionalParameters.java

```diff
@@ -0,0 +1,202 @@
+     * two byte integer values.
+     *
+     * @param responseData the Packet to read from.
+     * @param startPosition the read starting position.
+     * @return this object.
+     */
+    public static @Nullable FunctionalParameters readArray(Packet responseData, int startPosition) {
+        int pos = startPosition;
+        boolean isValid = false;
+        FunctionalParameters result = new FunctionalParameters();
+        for (int i = 0; i < FUNCTIONAL_PARAMETER_COUNT; i++) {
+            int value = responseData.getTwoByteValue(pos);
+            if (isNormalPosition(value)) {
+                result.values[i] = value;
+                isValid = true;
+            }
+            pos = pos + 2;
+        }
+        return isValid ? result : null;
+    }
+
+    /**
+     * Create a FunctionalParameters instance from the given Packet. Where the parameters are packed into an array of
+     * three byte records each comprising a one byte index followed by a two byte integer value.
+     *
+     * @param responseData the Packet to read from.
+     * @param startPosition the read starting position.
+     * @return this object.
+     */
+    public static @Nullable FunctionalParameters readArrayIndexed(Packet responseData, int startPosition) {
+        int pos = startPosition;
+        boolean isValid = false;
+        FunctionalParameters result = new FunctionalParameters();
+        for (int i = 0; i < FUNCTIONAL_PARAMETER_COUNT; i++) {
+            int index = responseData.getOneByteValue(pos) - 1;
+            pos++;
+            if ((index >= 0) && (index < FUNCTIONAL_PARAMETER_COUNT)) {
+                int value = responseData.getTwoByteValue(pos);
+                if (isNormalPosition(value)) {
+                    result.values[index] = value;
+                    isValid = true;
+                }
+            }
+            pos = pos + 2;
+        }
+        return isValid ? result : null;
+    }
+
+    /**
+     * Write the Functional Parameters to the given packet. Only writes normal valid position values.
+     *
+     * @param requestData the Packet to write to.
+     * @param startPosition the write starting position.
+     * @return fpIndex a bit map that indicates which of the written Functional Parameters contains a normal valid
+     *         position value.
+     */
+    public int writeArray(Packet requestData, int startPosition) {
+        int bitMask = 0b10000000;
+        int pos = startPosition;
+        int fpIndex = 0;
+        for (int i = 0; i < FUNCTIONAL_PARAMETER_COUNT; i++) {
+            if (isNormalPosition(values[i])) {
+                fpIndex |= bitMask;
+                requestData.setTwoByteValue(pos, values[i]);
+            }
+            pos = pos + 2;
+            bitMask = bitMask >>> 1;
+        }
+        return fpIndex;
+    }
+
+    @Override
+    public boolean equals(@Nullable Object obj) {
```

Comment:
When overriding `equals`, you must also override `hashCode`.

Reply:
Ok. Will do.

Reply:
Done